xdem/coreg/workflows.py

```diff
@@ -220,16 +224,28 @@ def dem_coregistration(
 
     if isinstance(ref_dem_path, str):
         if grid == "ref":
-            ref_dem, src_dem = gu.raster.load_multiple_rasters([ref_dem_path, src_dem_path], ref_grid=0)
+            ref_dem, src_dem = gu.raster.load_multiple_rasters([ref_dem_path, src_dem_path])
         elif grid == "src":
-            ref_dem, src_dem = gu.raster.load_multiple_rasters([ref_dem_path, src_dem_path], ref_grid=1)
+            ref_dem, src_dem = gu.raster.load_multiple_rasters([ref_dem_path, src_dem_path])
     else:
         ref_dem = ref_dem_path
         src_dem = src_dem_path
-        if grid == "ref":
-            src_dem = src_dem.reproject(ref_dem, silent=True)
-        elif grid == "src":
-            ref_dem = ref_dem.reproject(src_dem, silent=True)
+
+    # If an initial shift is provided, apply it before coregistration
+    if estimated_initial_shift:
+        logging.warning("Initial shift in affine mode only")
```

Comment:
Instead of this warning, should we rather add a raise in case we are not in an affine coregistration?